worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightPlatformAdapter.java

```diff
@@ -591,9 +603,27 @@ static void removeBeacon(BlockEntity beacon, LevelChunk levelChunk) {
 
     static List<Entity> getEntities(LevelChunk chunk) {
         if (PaperLib.isPaper()) {
-            return Arrays.asList(chunk.entities.getRawData());
+            if (POST_CHUNK_REWRITE) {
+                try {
+                    //noinspection unchecked
+                    return (List<Entity>) PAPER_CHUNK_GEN_ALL_ENTITIES.invoke(chunk.level.getEntityLookup().getChunk(chunk.locX, chunk.locZ));
+                } catch (IllegalAccessException | InvocationTargetException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+            try {
+                EntityList entityList = (EntityList) LevelChunk.class.getDeclaredField("entities").get(chunk);
+                return List.of(entityList.getRawData());
+            } catch (IllegalAccessException | NoSuchFieldException e) {
+                // fall through
+            }
+        }
+        try {
+            ((PersistentEntitySectionManager<Entity>) (ServerLevel.class.getDeclaredField("entityManager").get(chunk.level))).getEntities(chunk.getPos());
+        } catch (IllegalAccessException | NoSuchFieldException e) {
+            // fall through
         }
-        return chunk.level.entityManager.getEntities(chunk.getPos());
+        return List.of();
```

Comment:
Not sure if this is a good fallback, it should work on all platforms and ignoring all exceptions makes detecting bugs here very difficult.